src/json.c

```diff
@@ -64,31 +64,36 @@ json_print_escaped_string(str_buf_t *p_str_buf, const char *p_input_str)
     for (const char *in_ptr = p_input_str; '\0' != *in_ptr; ++in_ptr)
     {
         const char in_chr = *in_ptr;
+        bool       res    = false;
         switch (in_chr)
         {
             case '\\':
             case '\"':
-                str_buf_printf(p_str_buf, "\\%c", in_chr);
+                res = str_buf_printf(p_str_buf, "\\%c", in_chr);
                 break;
             case '\b':
-                str_buf_printf(p_str_buf, "\\b");
+                res = str_buf_printf(p_str_buf, "\\b");
                 break;
             case '\f':
-                str_buf_printf(p_str_buf, "\\f");
+                res = str_buf_printf(p_str_buf, "\\f");
                 break;
             case '\n':
-                str_buf_printf(p_str_buf, "\\n");
+                res = str_buf_printf(p_str_buf, "\\n");
                 break;
             case '\r':
-                str_buf_printf(p_str_buf, "\\r");
+                res = str_buf_printf(p_str_buf, "\\r");
                 break;
             case '\t':
-                str_buf_printf(p_str_buf, "\\t");
+                res = str_buf_printf(p_str_buf, "\\t");
                 break;
             default:
-                json_print_char_escaped(p_str_buf, in_chr);
+                res = json_print_char_escaped(p_str_buf, in_chr);
                 break;
         }
+        if (!res)
```

Comment:
It's hard to check this logic. Comment on `str_buf_printf` would lead me. to believe that we're checking for a null pointer, but type is a bool. https://github.com/ruuvi/ruuvi.esp_wrappers.c/issues/7

I'm assuming that functions above return `true` on success?